File: FCL/src/main/java/com/tungsten/fcl/ui/setting/AboutPage.java
package com.tungsten.fcl.ui.setting;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.tungsten.fcl.R;
import com.mio.util.AndroidUtilKt;
import com.tungsten.fclcore.task.Task;
import com.tungsten.fcllibrary.component.ui.FCLPage;

/**
 * 关于页：RecyclerView 行级复用，说明（about_desc）置顶，下方为链接行。
 */
public class AboutPage extends FCLPage {

    private static final int TYPE_DESC = 0;
    private static final int TYPE_LINK = 1;

    private static final String QQ_GROUP_KEY = "9_Mnxe5x1l6L7giLuRYQyBh0iWBgCUbw";

    /** 条目顺序：说明置顶，随后为各链接行 */
    private static final int[] TITLES = {
            R.string.about_desc,
            R.string.about_launcher,
            R.string.about_developer,
            R.string.community_discord,
            R.string.community_qq,
            R.string.about_sponsor,
            R.string.about_source
    };

    public AboutPage(Context context, int id, int resId) {
        super(context, id, resId);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        RecyclerView recyclerView = findViewById(R.id.about_list);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(new AboutAdapter());
    }

    @Override
    public Task<?> refresh(Object... param) {
        return null;
    }

    private class AboutAdapter extends RecyclerView.Adapter<AboutAdapter.Holder> {

        @Override
        public int getItemViewType(int position) {
            return position == 0 ? TYPE_DESC : TYPE_LINK;
        }

        @NonNull
        @Override
        public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            int layout = viewType == TYPE_DESC ? R.layout.item_about_desc : R.layout.item_about;
            return new Holder(LayoutInflater.from(parent.getContext()).inflate(layout, parent, false), viewType);
        }

        @Override
        public void onBindViewHolder(@NonNull Holder holder, int position) {
            holder.bind(position);
        }

        @Override
        public int getItemCount() {
            return TITLES.length;
        }

        class Holder extends RecyclerView.ViewHolder {
            private final TextView title;
            private final boolean link;

            Holder(@NonNull View itemView, int viewType) {
                super(itemView);
                title = itemView.findViewById(R.id.title);
                link = viewType == TYPE_LINK;
                if (link) {
                    itemView.setOnClickListener(v -> openLink(getBindingAdapterPosition()));
                }
            }

            void bind(int position) {
                title.setText(TITLES[position]);
            }
        }
    }

    private void openLink(int position) {
        switch (position) {
            case 1:
                AndroidUtilKt.openLink(getContext(), "https://fcl-team.github.io/");
                break;
            case 2:
                AndroidUtilKt.openLink(getContext(), "https://github.com/FCL-Team");
                break;
            case 3:
                AndroidUtilKt.openLink(getContext(), "https://discord.gg/ffhvuXTwyV");
                break;
            case 4:
                joinQQGroup(QQ_GROUP_KEY);
                break;
            case 5:
                AndroidUtilKt.openLink(getContext(), "https://afdian.com/@tungs");
                break;
            case 6:
                AndroidUtilKt.openLink(getContext(), "https://github.com/FCL-Team/FoldCraftLauncher");
                break;
        }
    }

    private void joinQQGroup(String key) {
        Intent intent = new Intent();
        intent.setData(Uri.parse("mqqopensdkapi://bizAgent/qm/qr?url=http%3A%2F%2Fqm.qq.com%2Fcgi-bin%2Fqm%2Fqr%3Ffrom%3Dapp%26p%3Dandroid%26jump_from%3Dwebapi%26k%3D" + key));
        try {
            getContext().startActivity(intent);
        } catch (Exception ignored) {
        }
    }
}

File: FCL/src/main/java/com/tungsten/fcl/ui/setting/AboutPage.kt
package com.tungsten.fcl.ui.setting

import android.content.Context
import android.content.Intent
import android.content.res.ColorStateList
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.mio.ui.adapter.SpacingItemDecoration
import com.mio.util.openLink
import com.tungsten.fcl.R
import com.tungsten.fcl.databinding.ItemAboutBinding
import com.tungsten.fcl.databinding.ItemAboutDescBinding
import com.tungsten.fcl.databinding.PageSettingAboutBinding
import com.tungsten.fclcore.task.Task
import com.tungsten.fcllibrary.component.theme.ThemeEngine
import com.tungsten.fcllibrary.component.ui.FCLPage

/**
 * 关于页：说明置顶，下方链接行合成一组（组内绘制分割线），
 * 行背景与分割线样式与版本设置页一致。
 */
class AboutPage(context: Context?, id: Int) : FCLPage(context, id, R.layout.page_setting_about) {

    private lateinit var binding: PageSettingAboutBinding

    init {
        create()
    }

    private fun create() {
        binding = PageSettingAboutBinding.bind(contentView)
        binding.aboutList.layoutManager = LinearLayoutManager(context)
        // 说明行与链接组之间留 8dp 间距，组内行间留 1dp 缝并绘制分割线
        val rowSpacing = (8 * context.resources.displayMetrics.density).toInt()
        val groupDivider = (1 * context.resources.displayMetrics.density).toInt()
        binding.aboutList.addItemDecoration(
            SpacingItemDecoration(
                rowSpacing,
                { parent, position ->
                    val adapter = parent.adapter as? AboutAdapter
                    if (adapter?.isNextInSameGroup(position) == true) groupDivider else rowSpacing
                },
                { ThemeEngine.getInstance().getTheme().getColor() }
            )
        )
        // 主题切换时重绘分割线颜色
        ThemeEngine.getInstance().registerEvent(binding.aboutList) {
            binding.aboutList.invalidate()
        }
        binding.aboutList.adapter = AboutAdapter { openLinkAt(it) }
    }

    override fun refresh(vararg param: Any?): Task<*>? {
        return null
    }

    private fun openLinkAt(linkIndex: Int) {
        val link = LINKS[linkIndex].link
        if (link == null) {
            joinQQGroup(QQ_GROUP_KEY)
        } else {
            openLink(context, link)
        }
    }

    private fun joinQQGroup(key: String) {
        val intent = Intent().apply {
            data = Uri.parse(
                "mqqopensdkapi://bizAgent/qm/qr?url=http%3A%2F%2Fqm.qq.com%2Fcgi-bin%2Fqm%2Fqr%3Ffrom%3Dapp%26p%3Dandroid%26jump_from%3Dwebapi%26k%3D" + key
            )
        }
        try {
            context.startActivity(intent)
        } catch (ignored: Exception) {
        }
    }

    /** 链接行：标题资源与跳转链接（null 表示跳转 QQ 群），条目顺序即显示顺序 */
    private data class LinkItem(val titleRes: Int, val link: String?)

    private class AboutAdapter(
        private val onLinkClick: (Int) -> Unit
    ) : RecyclerView.Adapter<AboutAdapter.Holder>() {

        override fun getItemCount(): Int = LINKS.size + 1

        override fun getItemViewType(position: Int): Int = if (position == 0) TYPE_DESC else TYPE_LINK

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
            val inflater = LayoutInflater.from(parent.context)
            val item = if (viewType == TYPE_DESC) {
                ItemAboutDescBinding.inflate(inflater, parent, false).root
            } else {
                ItemAboutBinding.inflate(inflater, parent, false).root
            }
            return Holder(item)
        }

        override fun onBindViewHolder(holder: Holder, position: Int) {
            // 行背景为圆角形状：链接行合成一组（组首上圆角/组尾下圆角/中间无圆角），
            // 颜色 tint 取主题浅色；组内行间缝隙绘制分割线
            holder.itemView.setBackgroundResource(
                when (position) {
                    0 -> R.drawable.bg_item_rounded
                    1 -> R.drawable.bg_item_rounded_top
                    itemCount - 1 -> R.drawable.bg_item_rounded_bottom
                    else -> R.drawable.bg_item_rounded_middle
                }
            )
            ThemeEngine.getInstance().unregisterEvent(holder.itemView)
            ThemeEngine.getInstance().registerEvent(holder.itemView) {
                holder.itemView.backgroundTintList =
                    ColorStateList.valueOf(ThemeEngine.getInstance().getTheme().ltColor)
            }
            if (position == 0) {
                holder.itemView.setOnClickListener(null)
                ItemAboutDescBinding.bind(holder.itemView).title.setText(R.string.about_desc)
            } else {
                holder.itemView.setOnClickListener {
                    onLinkClick(holder.bindingAdapterPosition - 1)
                }
                ItemAboutBinding.bind(holder.itemView).title.setText(LINKS[position - 1].titleRes)
            }
        }

        /** 供间距装饰器判断：链接行内相邻时行间留 1dp 缝（绘制分割线） */
        fun isNextInSameGroup(position: Int): Boolean =
            position > 0 && position + 1 < itemCount

        class Holder(itemView: View) : RecyclerView.ViewHolder(itemView)
    }

    companion object {
        private const val TYPE_DESC = 0
        private const val TYPE_LINK = 1
        private const val QQ_GROUP_KEY = "9_Mnxe5x1l6L7giLuRYQyBh0iWBgCUbw"

        private val LINKS = listOf(
            LinkItem(R.string.about_launcher, "https://fcl-team.github.io/"),
            LinkItem(R.string.about_developer, "https://github.com/FCL-Team"),
            LinkItem(R.string.community_discord, "https://discord.gg/ffhvuXTwyV"),
            LinkItem(R.string.community_qq, null),
            LinkItem(R.string.about_sponsor, "https://afdian.com/@tungs"),
            LinkItem(R.string.about_source, "https://github.com/FCL-Team/FoldCraftLauncher")
        )
    }
}